accepted/2020/windows-specific-apis/windows-specific-apis.md

```diff
@@ -0,0 +1,475 @@
+# Marking Windows-only APIs
+
+As discussed in [dotnet/runtime#33331] we're adding an analyzer to detect usage
+of platform-specific APIs. This analyzer requires a custom attribute applied to
+method, containing type, or assembly.
+
+In this proposal I'm covering which APIs we mark as Windows-specific an how. It
+was largely informed by the work put into [platform-compat].
+
+[dotnet/runtime#33331]: https://github.com/dotnet/runtime/issues/33331
+[platform-compat]: https://github.com/dotnet/platform-compat
+
+## Which Windows version will we use?
+
+The lowest version of Windows that we support with .NET Core is Windows 7. Also,
+we generally don't expose functionality that requires a higher version of
+Windows.
+
+Thus, I propose that we use `7.0` as the minimum version number instead of
+hunting down which Windows version actually introduced a given feature, which
+means the custom attribute will look as follows:
+
+```C#
+[MinimumOSPlatform("windows7.0")]
+```
+
+## Entire assemblies
+
+There is a set of assemblies that we can mark wholesale as being
+Windows-specific:
+
+> **OPEN**: Should `System.Threading.Overlapped.dll` be removed?
+> **OPEN**: I was told that part of `DirectoryServices` is now cross-platform. Is that true?
+
+* **Microsoft.Win32.Registry.dll**
+* **Microsoft.Win32.Registry.AccessControl.dll**
+* **Microsoft.Win32.SystemEvents.dll**
+* **System.Diagnostics.EventLog.dll**
+* **System.Diagnostics.PerformanceCounter.dll**
+* **System.DirectoryServices.dll**
+* **System.DirectoryServices.AccountManagement.dll**
+* **System.DirectoryServices.Protocols.dll**
+* **System.IO.FileSystem.AccessControl.dll**
+* **System.IO.FileSystem.DriveInfo.dll**
+* **System.IO.Pipes.AccessControl.dll**
+* **System.Management.dll**
+* **System.Net.Http.WinHttpHandler.dll**
+* **System.Runtime.InteropServices.WindowsRuntime.dll**
+* **System.Security.AccessControl.dll**
+* **System.Security.Cryptography.Cng.dll**
+* **System.Security.Cryptography.ProtectedData.dll**
+* **System.Security.Principal.Windows.dll**
+* **System.ServiceProcess.ServiceController.dll**
+* **System.Threading.AccessControl.dll**
```

Comment:
we don't ship a 5.0 package, does it mean that this assembly should not be annotated with the new attribute?

https://github.com/dotnet/runtime/blob/f9fc9fc2433a35c2dc5dc2ba1d62136e7e682be4/src/libraries/System.Threading.AccessControl/src/System.Threading.AccessControl.csproj#L3

Reply:
We can use an internal attribute for this one too.